### PR TITLE
Check ERC20 address matches registered address

### LINKED ISSUE #954: Deposit function does not check if ERC20 address matches that of registered asset definition
[Zulip thread](https://espresso.zulipchat.com/#narrow/stream/311688-CAPE/topic/depositErc20.20address)

The explicit check (1.) is absent in the contract:

https://github.com/EspressoSystems/cape/blob/16e7fa0573e2a95ff9d089e1d7895d595775cbb6/doc/workflow/src/lib.rs#L236-L244

### PR DESCRIPTION
**Pull request by sveitser**
- Add test for deposit with wrong ERC20 address
- Some refactoring to make the test code more concise.

Close #954

### CODE REVIEW

**Review by jbearer — APPROVED** [review]

File: contracts/contracts/CAPE.sol [hunk 1]
```diff
@@ -213,6 +213,7 @@ contract CAPE is RecordsMerkleTree, RootStore, AssetRegistry, ReentrancyGuard {
     /// @param erc20Address address of the ERC-20 token corresponding to the deposit
     function depositErc20(RecordOpening memory ro, address erc20Address) public nonReentrant {
```
Comment (alxiong) [inline contracts/contracts/CAPE.sol:214]: Shall we remove the `address erc20Address` in this PR?

Comment (philippecamacho) [reply contracts/contracts/CAPE.sol:214]: We prefer not to for now to avoid breaking the ui code. @sveitser will create another ticket to do this later.

**Review by philippecamacho — APPROVED** [review]
LGTM

File: contracts/contracts/CAPE.sol (hunk 1 above)
Comment (sveitser) [reply contracts/contracts/CAPE.sol:214]: The ticket is #971

**Pull request merged**